### PR TITLE
[ACS-8664] generic question redirection to hx insight page

### PR DESCRIPTION
**Pull request by AleksanderSklorz**
**JIRA ticket link or changeset's description**
https://hyland.atlassian.net/browse/ACS-8664

ADF PR: https://github.com/Alfresco/alfresco-ng2-components/pull/10174

### CODE REVIEW

**Review by g-jaskowski — APPROVED** [review]

File: projects/aca-content/src/lib/components/knowledge-retrieval/search-ai/agents-button/agents-button.component.spec.ts [hunk 1]
```diff
@@ -169,55 +206,87 @@ describe('AgentsButtonComponent', () => {
...
-        });
+            getAgentsButton().dispatchEvent(event);
+            fixture.detectChanges();
+            const menuTrigger = getMenuTrigger();
+            expect(menuTrigger).toBeTruthy();
+            expect(menuTrigger).toBe(getMenu());
+          });
+
+          it('should call checkSearchAvailability from SearchAiService with correct parameter', () => {
+            fixture.detectChanges();
```
Comment (nikita-web-ua) [inline projects/aca-content/src/lib/components/knowledge-retrieval/search-ai/agents-button/agents-button.component.spec.ts:243]: can we move  `fixture.detectChanges();` to beforeEach?

File: projects/aca-content/src/lib/components/knowledge-retrieval/search-ai/agents-button/agents-button.component.ts [hunk 2]
```diff
@@ -84,13 +85,18 @@ export class AgentsButtonComponent implements OnInit, OnDestroy {
       .subscribe((selection) => {
         this.selectedNodesState = selection;
       });
-
-    this.agentService
-      .getAgents()
+    forkJoin({
+      agents: this.agentService.getAgents().pipe(
+        take(1),
```
[the remark this answers is not in the document]
Comment (AleksanderSklorz) [reply projects/aca-content/src/lib/components/knowledge-retrieval/search-ai/agents-button/agents-button.component.ts:90]: yes, I tried without it but then forkJoin is not triggered so we need it

File: projects/aca-content/src/lib/components/knowledge-retrieval/search-ai/agents-button/agents-button.component.spec.ts (hunk 1 above)
Comment (AleksanderSklorz) [reply projects/aca-content/src/lib/components/knowledge-retrieval/search-ai/agents-button/agents-button.component.spec.ts]: done

File: projects/aca-content/src/lib/components/knowledge-retrieval/search-ai/agents-button/agents-button.component.ts [hunk 3]
```diff
@@ -84,13 +85,18 @@ export class AgentsButtonComponent implements OnInit, OnDestroy {
...
+        catchError(() => throwError('KNOWLEDGE_RETRIEVAL.SEARCH.ERRORS.AGENTS_FETCHING'))
+      ),
+      config: this.searchAiService.getConfig().pipe(catchError(() => throwError('KNOWLEDGE_RETRIEVAL.SEARCH.ERRORS.HX_INSIGHT_URL_FETCHING')))
+    })
       .pipe(takeUntil(this.onDestroy$))
```
[the remark this answers is not in the document]
Comment (AleksanderSklorz) [reply projects/aca-content/src/lib/components/knowledge-retrieval/search-ai/agents-button/agents-button.component.ts]: done

File: projects/aca-content/src/lib/components/knowledge-retrieval/search-ai/agents-button/agents-button.component.spec.ts [hunk 4]
```diff
@@ -169,55 +186,119 @@ describe('AgentsButtonComponent', () => {
...
+            expect(getMenuTrigger()).toBeNull();
+          });
+
+          it('should enable menu triggering if checkSearchAvailability from SearchAiService returns empty message', () => {
+            checkSearchAvailabilitySpy.and.returnValue('');
+            fixture.detectChanges();
+
+            getAgentsButton().dispatchEvent(event);
+            fixture.detectChanges();
```
[the remark this answers is not in the document]
Comment (AleksanderSklorz) [reply projects/aca-content/src/lib/components/knowledge-retrieval/search-ai/agents-button/agents-button.component.spec.ts:269]: I checked it, seems like it is needed after dispatchEvent but I removed it from before dispatchEvent so now it's called only once

File: projects/aca-content/src/lib/components/knowledge-retrieval/search-ai/agents-button/agents-button.component.ts [hunk 5]
```diff
@@ -111,11 +117,16 @@ export class AgentsButtonComponent implements OnInit, OnDestroy {
   }
 
   onClick(): void {
-    const error = this.searchAiService.checkSearchAvailability(this.selectedNodesState);
-    if (error) {
-      this.notificationService.showInfo(error);
+    if (this.selectedNodesState.isEmpty) {
```
[the remark this answers is not in the document]
Comment (AleksanderSklorz) [reply projects/aca-content/src/lib/components/knowledge-retrieval/search-ai/agents-button/agents-button.component.ts]: Done, for second part I now hide this button similarly like it is done when agents are missing

**Review by nikita-web-ua — APPROVED** [review]

**Pull request merged**